db/records/operations/select.py

```diff
@@ -48,7 +46,32 @@ def list_records_from_table(
         json.dumps(order) if order is not None else None,
         json.dumps(filter) if filter is not None else None,
         json.dumps(group) if group is not None else None,
-        json.dumps(search) if search is not None else None,
+    ).fetchone()[0]
+    return result
+
+
+def search_records_from_table(
+        conn,
+        table_oid,
+        search=[],
+        limit=10,
+):
+    """
+    Get records from a table, according to a search specification
+
+    Only data from which the user is granted `SELECT` is returned.
+
+    Args:
+        tab_id: The OID of the table whose records we'll get.
+        search: A list of dictionaries defining a search.
+        limit: The maximum number of rows we'll return.
+
+    The search definition objects should have the form
+    {"attnum": <int>, "direction": <text>}
+    """
+    search = search or []
+    result = db_conn.exec_msar_func(
+        conn, 'search_records_from_table', table_oid, json.dumps(search), limit
```

Comment:
We could, but I'd rather have the front end do that. I.e., if they want to avoid the call, they should just not send the request until there are letters in the array. For the completeness of this function I'd really rather treat it as a kind of "narrowing filter" concept. And for that, we should return unfiltered results whenever we get an empty search definition.